### PR TITLE
build: add pipeline status Execute method scaffolding

### PR DESCRIPTION
**Pull request by huanjani**
Built up the pipeline status subcommand by adding **scaffolding** for: the Execute method in the cli package (and relevant test in test file) and the Describe method in the describe package.

Partially addresses #928

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/pipeline_status.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+// BuildPipelineStatusCmd builds the command for showing the status of a deployed pipeline.
+func BuildPipelineStatusCmd() *cobra.Command {
+	vars := pipelineStatusVars{
+		GlobalOpts: NewGlobalOpts(),
+	}
+	cmd := &cobra.Command{
+		Hidden: true, // TODO: remove when ready for production!
+		Use:    "status",
+		Short:  "Show the status of your application's pipeline.",
+		Long:   "Show the status of each deployed pipeline's stages.",
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_status.go:65]:
```suggestion
		Long:   "Show the status of each stage in your pipeline.",
```

File: internal/pkg/cli/pipeline_status.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+// BuildPipelineStatusCmd builds the command for showing the status of a deployed pipeline.
+func BuildPipelineStatusCmd() *cobra.Command {
+	vars := pipelineStatusVars{
+		GlobalOpts: NewGlobalOpts(),
+	}
+	cmd := &cobra.Command{
+		Hidden: true, // TODO: remove when ready for production!
+		Use:    "status",
+		Short:  "Show the status of your application's pipeline.",
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_status.go:64]:
```suggestion
		Short:  "Show the status of a pipeline.",
```

File: internal/pkg/cli/pipeline_status.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		Long:   "Show the status of each deployed pipeline's stages.",
+
+		Example: `
+Shows status of the deployed pipeline "pipeline-myapp-mycompany-myrepo".
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_status.go:68]:
```suggestion
Shows status of the pipeline "pipeline-myapp-myrepo".
```

File: internal/pkg/cli/pipeline_status.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+/code $ copilot pipeline status -n pipeline-myapp-mycompany-myrepo`,
```
Comment (kohidave) [inline internal/pkg/cli/pipeline_status.go:69]:
```suggestion
/code $ copilot pipeline status -n pipeline-myapp-myrepo `,
```

File: internal/pkg/cli/pipeline_status.go (hunk 1 above)
Comment (huanjani) [reply internal/pkg/cli/pipeline_status.go:65]: Ahh, much better. Thanks! But "stage IN" or "stage OF"?

File: internal/pkg/cli/pipeline_status.go [hunk 5]
```diff
@@ -47,12 +61,12 @@ func BuildPipelineStatusCmd() *cobra.Command {
...
 		Use:    "status",
-		Short:  "Show the status of your application's pipeline.",
+		Short:  "Show the status of a pipeline.",
 		Long:   "Show the status of each deployed pipeline's stages.",
 
 		Example: `
-Shows status of the deployed pipeline "pipeline-myapp-mycompany-myrepo".
-/code $ copilot pipeline status -n pipeline-myapp-mycompany-myrepo`,
+Shows status of the pipeline "pipeline-myapp-myrepo".
+/code $ copilot pipeline status -n pipeline-myapp-myrepo`,
```
Comment (SoManyHs) [inline internal/pkg/cli/pipeline_status.go:69]: any reason why we're taking out "mycompany"?

File: internal/pkg/cli/pipeline_status.go [hunk 6]
```diff
@@ -36,6 +41,15 @@ func (o *pipelineStatusOpts) Ask() error {
 
 // Execute displays the status of the pipeline.
 func (o *pipelineStatusOpts) Execute() error {
+	err := o.initStatusDescriber(o)
+	if err != nil {
+		return fmt.Errorf("describe status of pipeline : ")
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_status.go:46]: do we need to put the error here?
```suggestion
		return fmt.Errorf("describe status of pipeline: %w", err)
```

File: internal/pkg/describe/pipeline_status.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import ()
```
Comment (efekarakus) [inline internal/pkg/describe/pipeline_status.go:6]: ? what happened here :D

File: internal/pkg/cli/pipeline_status.go [hunk 8]
```diff
@@ -36,6 +41,15 @@ func (o *pipelineStatusOpts) Ask() error {
 
 // Execute displays the status of the pipeline.
 func (o *pipelineStatusOpts) Execute() error {
+	err := o.initStatusDescriber(o)
+	if err != nil {
+		return fmt.Errorf("describe status of pipeline: %w", err)
+	}
+	pipelineStatus, err := o.statusDescriber.Describe()
+	if err != nil {
+		return fmt.Errorf("describe status of pipeline : ")
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_status.go:50]: same here

File: internal/pkg/describe/pipeline_status.go (hunk 7 above)
Comment (huanjani) [reply internal/pkg/describe/pipeline_status.go:6]: Haha, you mean the total absence of imports? The whole file is a bare-bones skeleton, so there was nothing to import (and wouldn't the compiler yell at me if I imported something and it went unused?). Is there something I should do differently in this situation? Maybe not even bother committing this file like this?

File: internal/pkg/cli/pipeline_status.go (hunk 5 above)
Comment (huanjani) [reply internal/pkg/cli/pipeline_status.go:69]: @kohidave

File: internal/pkg/describe/pipeline_status.go (hunk 7 above)
Comment (efekarakus) [reply internal/pkg/describe/pipeline_status.go:6]: nono i mean we can just delete this line and it should still work

File: internal/pkg/cli/interfaces.go [hunk 9]
```diff
@@ -308,6 +308,10 @@ type envDescriber interface {
 	Describe() (*describe.EnvDescription, error)
 }
 
+type pipelineStatusDescriber interface {
```
[the remark this answers is not in the document]
Comment (huanjani) [reply internal/pkg/cli/interfaces.go:311]: Sounds dreamy indeed! I'll see if I can figure out how to use that instead in the next PR...!

File: internal/pkg/cli/pipeline_status.go (hunk 5 above)
Comment (kohidave) [reply internal/pkg/cli/pipeline_status.go:69]: just to keep the suggested lines a little short. i don't have any great reasoning behind it other than i just thought it looked less cluttered

File: internal/pkg/cli/pipeline_status.go (hunk 1 above)
Comment (kohidave) [reply internal/pkg/cli/pipeline_status.go:65]: Oh maybe of sounds better? Either's fine with me! Your choice.

File: internal/pkg/cli/pipeline_status.go (hunk 8 above)
Comment (kohidave) [inline internal/pkg/cli/pipeline_status.go:50]:
```suggestion
		return fmt.Errorf("describe status of pipeline : %w", err)
```

File: internal/pkg/describe/pipeline_status.go (hunk 7 above)
Comment (kohidave) [inline internal/pkg/describe/pipeline_status.go:6]:
```suggestion
```

Comment (huanjani) [reply internal/pkg/describe/pipeline_status.go:6]: Ah, I see. Will delete!

File: internal/pkg/cli/pipeline_status.go [hunk 10]
```diff
@@ -36,6 +41,15 @@ func (o *pipelineStatusOpts) Ask() error {
 
 // Execute displays the status of the pipeline.
 func (o *pipelineStatusOpts) Execute() error {
+	err := o.initStatusDescriber(o)
+	if err != nil {
+		return fmt.Errorf("describe status of pipeline: %w", err)
+	}
+	pipelineStatus, err := o.statusDescriber.Describe()
+	if err != nil {
+		return fmt.Errorf("describe status of pipeline : %w", err)
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_status.go:50]:
```suggestion
		return fmt.Errorf("describe status of pipeline: %w", err)
```
There is an extra space here :P

File: internal/pkg/cli/pipeline_status.go [hunk 11]
```diff
@@ -47,12 +61,12 @@ func BuildPipelineStatusCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Hidden: true, // TODO: remove when ready for production!
 		Use:    "status",
-		Short:  "Show the status of your application's pipeline.",
-		Long:   "Show the status of each deployed pipeline's stages.",
+		Short:  "Show the status of a pipeline.",
+		Long:   "Show the status of each stage of your pipeline.",
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_status.go:65]:
```suggestion
		Short:  "Shows the status of a pipeline.",
		Long:   "Shows the status of each stage of your pipeline.",
```
Show**s** to be consistent with other "show" commands

**Pull request merged**